src/librustc_back/target/mod.rs

```diff
@@ -698,6 +707,10 @@ impl ToJson for Target {
         target_option_val!(max_atomic_width);
         target_option_val!(panic_strategy);
 
+        if self.options.max_atomic_width.to_string() != self.target_pointer_width {
```

Comment:
This was the simplest change necessary to make this work but using `Option` here sounds better; I'll give it a try.